extensions/azdata/src/test/common/httpClient.test.ts

```diff
@@ -45,14 +45,30 @@ describe('HttpClient', function () {
 		await should(downloadPromise).be.rejected();
 	});
 
+	it('rejects on non-OK status code', async function (): Promise<void> {
+		const downloadPath = path.join(os.tmpdir());
+		nock('https://127.0.0.1')
+			.get('/')
+			.reply(404, '');
+		const downloadPromise = HttpClient.download('https://127.0.0.1', downloadPath, outputChannelMock.object);
+
+		await should(downloadPromise).be.rejected();
+	});
+
 	it('errors on write stream error', async function (): Promise<void> {
-		const downloadPath = path.join(os.tmpdir(), `azdata-httpClientTest-error-${uuid.v4()}.txt`);
+		const downloadPath = path.join(os.tmpdir());
```

Comment:
What is the purpose of joining just one path? Perhaps a comment will help clarify.

Reply:
No reason, missed cleaning that up.